### ISSUE TITLE
[package] gtest/x.y.z: Always appends library 'pthread' for Linux and FreeBSD

### ISSUE THREAD

**Issue by PiJ82** [conan-io/conan-center-index#19113]
### Description

Hey everyone,

in the `package_info()` method of the gtest conanfile, `pthread` is always appended as system library for Linux and FreeBSD.
```
def package_info(self):
[...]
    if self.settings.os in ["Linux", "FreeBSD"]:
            self.cpp_info.components["libgtest"].system_libs.append("m")
            self.cpp_info.components["libgtest"].system_libs.append("pthread")
```
See [here](https://github.com/conan-io/conan-center-index/blob/677f654558c98a746ac69c0c06c83fb4b7be67e1/recipes/gtest/all/conanfile.py#L161).

This may not be correct in case the gtest package was created with `-o gtest:disable_pthreads=True`.

Would it make sense to add an if-statement guarding this? E.g.:
```
def package_info(self):
[...]
    if self.settings.os in ["Linux", "FreeBSD"]:
            self.cpp_info.components["libgtest"].system_libs.append("m")
            if not.self.options.disable_pthreads:
                self.cpp_info.components["libgtest"].system_libs.append("pthread")
```

### Package and Environment Details

* Package Name/Version: **gtest/x.y.z**
* Operating System+version: **Linux Ubuntu 20.04**
* Compiler+version: **GCC 7**
* Conan version: **conan 1.60.2**
* Python version: **Python 3.10.5**

### Conan profile

[settings]
os=Linux
arch=x86_64
compiler=gcc
compiler.version=7
compiler.libcxx=libstdc++11
build_type=Release
[options]
gtest:disabe_pthreads=True
[conf]
[build_requires]
[env]

### Steps to reproduce

-

### Logs

-

**Comment by SpaceIm** [comment]
> Would it make sense to add an if-statement guarding this?

It totally makes sense. It has been forgotten when this option has been added in https://github.com/conan-io/conan-center-index/pull/17614

**Comment by RubenRBS** [comment]
As @SpaceIm says, your suggestion is totally valid, great catch! Feel free to open a PR with it if you want, else I'm sure someone else will pick the change up :)

**Comment by PiJ82** [comment]
I'll create a PR the next days. Thanks for the confirmation :)

**Comment by lkotsonis** [comment]
@PiJ82 did you open a PR for this after all? I can't seem to find one

**Comment by lkotsonis** [comment]
This issue was addressed with #23640 and it can be closed.

**Comment by uilianries** [comment]
@lkotsonis Thank you for reporting!
